edgar-doctor/src/components/app/dashboardPages/patients/subViews/PatientsSubViewAppointments.tsx

```diff
@@ -0,0 +1,88 @@
+'use client';
+
+import { useState } from 'react';
+
+import { Button, HStack, VStack } from '@chakra-ui/react';
+
+import paginationHandler from 'utils/navigation/paginationHandler';
+import Pagination from 'components/navigation/Pagination';
+import countMaxNumberPage from 'utils/navigation/countMaxNumberPage';
+
+import { type PatientType } from 'types/app/dashboard/patients/PatientType';
+import { useGetDoctorAppointmentsQuery } from 'services/request/appointments';
+import AppointmentCard from '../../appointments/AppointmentCard';
+
+const PatientsSubViewAppointments = ({
+	selectedPatient,
+}: {
+	selectedPatient: PatientType | undefined;
+}): JSX.Element => {
+	const [pageIndexNext, setPageIndexNext] = useState(1);
+	const [pageIndexOld, setPageIndexOld] = useState(1);
+	const [appointmentType, setAppointmentType] = useState<'upcoming' | 'past'>('upcoming');
+	const { data: appointments } = useGetDoctorAppointmentsQuery();
+	const patientAppointments = appointments
+		?.filter((appointment) => appointment.patientId === selectedPatient?.id)
+		?.sort((a, b) => new Date(a.startDate).getTime() - new Date(b.startDate).getTime());
+	const upcomingAppointments = patientAppointments?.filter(
+		(patientAppointment) =>
+			patientAppointment.endDate > new Date().getTime() &&
+			!patientAppointment.appointmentStatus.includes('CANCELED'),
+	);
+	const pastAppointments = patientAppointments?.filter(
+		(patientAppointment) =>
+			patientAppointment.endDate <= new Date().getTime() &&
+			!patientAppointment.appointmentStatus.includes('CANCELED'),
+	);
+
+	return (
+		<>
+			<HStack w="100%" spacing="8px">
+				<Button w="100%" size="md" variant="primary" onClick={() => setAppointmentType('upcoming')}>
+					Prochain rendez-vous
+				</Button>
+				<Button w="100%" size="md" variant="secondary" onClick={() => setAppointmentType('past')}>
```

Comment:
You need to change the variant of the button according to the value of the `appointmentType`